catt/subs_info.py

```diff
@@ -0,0 +1,51 @@
+import re
+
+import requests
+
+from .error import SubtitlesError
+from .util import create_temp_file
+
+
+class SubsInfo:
+    def __init__(self, subs_url: str, local_ip: str, port: int) -> None:
+        self._subs_url = subs_url
+        self._local_ip = local_ip
+        self.port = port
+
+        ext = subs_url.lower().split(".")[-1]
+        if "://" in subs_url:
```

Comment:
I think this could be a bit more elegant like so:

```
# Read subs (local or remote) into memory.
subs = read_subs(subs_url)

# If the file is in the SRT format, convert to WebVTT.
if ext == "srt":
    subs = self._convert_srt_to_webvtt(subs)

# Write the temp file with the subs.
self.file = create_temp_file(subs)
```

We're always writing the temp file in this case, but I think that's worth it to simplify the code. `read_subs` will either fetch the remote subs or read local subs, basically `_fetch_remote_subs` or `_read_srt_subs` depending on what's going on in the URL.